src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -1672,6 +1672,11 @@ public static IEnumerable<SyntaxToken> ParseTokens(string text, int offset = 0,
             }
         }
 
+        public static SyntaxTokenParser CreateTokenParser(SourceText sourceText, CSharpParseOptions? options)
```

Comment:
i have mixed feelings here.  i would prefer this align with SyntaxFactory.ParseTokens.  That means making options optional.  

i'm on the fence too about taking in an initialTokenPosition.  but i can live with leaving that out.

Reply:
I initially left options optional, but that actually caused a bug in my prototype implementation on the razor side. I'm not hugely opposed to making it optional (as I have now revealed the bug), but that's why I changed it.

I don't think an initial position is necessary. We have a skip method, if you need to change where you start, the tools are there.

Reply:
so, not pushing on this, but the general idea here is:

1. i as a caller have a chunk of text.   (but which of course then starts at pos 0)
2. that chunk of text represents a location in some larger text construct.
3. when i want to lex tokens in that subtext, i want to their positions to be accurate in the larger text.

but, given teh ability to skip, maybe this is ok.  It's also something we can wait on till the future.

Reply:
> that chunk of text represents a location in some larger text construct.

Again, this _isn't_ a goal of this API. This method takes a SourceText very specifically. The goal of this API is that you can parse C# tokens from a given source text, with accurate positions inside that source text. If you want to do some futher mapping where you created a source text from a smaller chunk of the original, that's fine, but that additional mapping is on the consumer of the API.